src/executor/host.rs

```diff
@@ -2389,21 +2385,14 @@ impl SignatureVerification {
 
                 // signature (64 bytes) + recovery ID (1 byte)
                 let sig_bytes = self.signature();
-                if let Ok(sig) =
-                    libsecp256k1::Signature::parse_standard_slice(&sig_bytes.as_ref()[..64])
-                {
-                    if let Ok(ri) = libsecp256k1::RecoveryId::parse(sig_bytes.as_ref()[64]) {
-                        if let Ok(actual) = libsecp256k1::recover(&message, &sig, &ri) {
-                            self.public_key().as_ref()[..] == actual.serialize_compressed()[..]
-                        } else {
-                            false
-                        }
-                    } else {
-                        false
-                    }
-                } else {
-                    false
-                }
+                libsecp256k1::Signature::parse_standard_slice(&sig_bytes.as_ref()[..64])
```

Comment:
this have no logic change, just my attempt to improve the readability of the code